### ISSUE TITLE
Learning Rate returns to default when loading model

### ISSUE THREAD

**Issue by jturner116** [dmlc/xgboost#8113]
In XGBoost 1.6.1:

I have trained a very normal XGBClassifier here: 
![image](https://user-images.githubusercontent.com/59941858/180487661-d9527a5e-9052-48e3-b73f-f57f274dec6b.png)

I have saved the model with .save_model. When loading the model, the learning rate has changed back to default even though the other hyperparameters stay the same. 
![image](https://user-images.githubusercontent.com/59941858/180487992-8ee41225-0d06-4ff9-9756-e0c5411b98f1.png)

This doesn't seem intended, but it is a good caution to me to not retrain a model I have saved hah. I can see this impacting model continuation after a save though.

Thanks!

**Comment by trivialfis** [comment]
Thank you for raising the issue. I will try to reproduce it.

**Comment by trivialfis** [comment]
Hi, I can't reproduce the issue:

``` python
from sklearn.datasets import make_classification
from xgboost import XGBClassifier

X, y = make_classification(random_state=0)
clf = XGBClassifier(learning_rate=0.79)
clf.fit(X, y)
print(clf)

clf.save_model("model")

clf = XGBClassifier()
clf.load_model("model")
print(clf)
assert clf.learning_rate == 0.79

```
Could you please share a reproducible example that I can run?

**Comment by jturner116** [comment]
`from sklearn.datasets import make_classification
from xgboost import XGBClassifier

X, y = make_classification(random_state=0)
clf = XGBClassifier(eta=0.79)
clf.fit(X, y)
print(clf)

clf.save_model("model")

clf = XGBClassifier()
clf.load_model("model")
print(clf)`

You are quite right, I should have made more clear that the issue seems connected to eta. This should produce the odd behavior. The same issue (I think) is addressed [here](https://datascience.stackexchange.com/questions/33885/eta-and-learning-rate-different-in-xgboost). Is it intended? Apologies if it has already been addressed. 

Thanks!

**Comment by trivialfis** [comment]
Hi, please use `learning_rate` for the sklearn interface. `learning_rate` is defined at the constructor so that sklearn base estimator can inspect the parameters and do its magic, while the alias `eta` is not defined there (`eta` is exactly the same as `learning_rate`, remained here for historical reasons).
